parser/Builder.cc

```diff
@@ -8,6 +8,7 @@
 #include "ruby_parser/diagnostic.hh"
 
 #include <algorithm>
+#include <map>
```

Comment:
Instead of std::map, can you rewrite this code to use `UnorderedMap`, which is a type alias that we've defined in `common.h`?

https://github.com/iliabylich/sorbet/blob/6499a58a7d8919a43f52d45a10a2d93af9d1d95e/common/common.h#L32

You should mostly be able to use the exact same APIs, but just will need to change the `include` and the method signatures.